core/src/utils/options.rs

```diff
@@ -30,6 +30,7 @@ pub struct SP1CoreOpts {
     pub prove_stream_capacity: usize,
     pub split_opts: SplitOpts,
     pub reconstruct_commitments: bool,
+    pub event_counts: EventCounts,
```

Comment:
why is event counts in sp1 core opts?

Reply:
this feels wrong, you can just put it into ExecutionState

Reply:
oh hm maybe its cause you need opts in new() in Runtime?

Reply:
There are a few reasons why it's not part of the state:

1) In the runtime recover function:  https://github.com/succinctlabs/sp1/blob/545ab109a75ccf942ba78b43014fe2c4928609d3/core/src/runtime/mod.rs#L242, the state is assigned after the creation of the runtime.
2) Semantically, it seems like the ExecutionState stores information about the runtime state as opposed to statistics of it.  So you could argue it should be a separate object.